src/plugins/data_source/common/data_sources/types.ts

```diff
@@ -14,6 +14,7 @@ export interface DataSourceAttributes extends SavedObjectAttributes {
     credentials: UsernamePasswordTypedContent | SigV4Content | undefined | AuthTypeContent;
   };
   lastUpdatedTime?: string;
+  name: AuthType | string;
```

Comment:
what's the difference of name vs auth.type?

Reply:
Name can be any unique string to identify the authentication method. 

auth.type is to specify using which method request needs to be signed. Right now we only have three ways: No auth, Basic authentication and AWS SigV4 authentication to sign the request. 

So for example, I want to add new authentication method named `abc` for role based authentication. So my plugin will do all the work such as assume role etc to get temporary credentials (Access key, secret key and session token) and provide those data source plugin and will ask to sign request using AWS SigV4 with provided credentials.

Reply:
I see, just curious not a blocker, do we show name on the UI or is it solely used as identifier internally?